### ISSUE TITLE
It's not DRY, it's kinda WET...

### ISSUE THREAD

**Issue by jerguslejko** [kristiankostecky/sort-it-out#8]
Heya 👨!

Although your code is now *MUCH* better then in was in terms of design, it still could be improved.

*DRY - Don't Repeat Yourself* is a rule that will be mentioned over and over and over...

There is *a thing* that repeated multiple times in there...

```php

function listNames($gender, $people) {
    $names = [];
    foreach($people as $person) {
        if ($person['gender'] == $gender) {
            $names[] = $person['name'];
        }
    }
    return implode(', ', $names);
};

function countGender($gender, $people) {
    $count = 0;
    foreach($people as $person) {
        if ($person['gender'] == $gender) {
            $count++;
        }
    }
    return $count;
};
```

Can you spot the repeated part?

```php
if ($person['gender'] == $gender) {
    ...
}
```

*OK*, but how do we get rid of it? By extracting it!

Instead of passing `$people` to each function, you can pass `$male` or `$female` directly, and get rid of the conditional. Just create a function that filters all males/females and returns them. And then pass them directly to according functions 🌵

**Comment by kristiankostecky** [comment]
Something like this? 🍰 
`$male = [];
$female = [];

foreach($data as $person) {
    if ($person['gender'] == 'male') {
        $male[] = $person;
    } else {
        $female[] = $person;
    }
}`

**Comment by jerguslejko** [comment]
Something like that.. Try to implement it as a function :cactus:

**Comment by jerguslejko** [comment]
Basically, I want you to edit your code like this:

Current solution:

```php
function listNames($gender, $people) {
    $names = [];
    foreach($people as $person) {
        if ($person['gender'] == $gender) {
            $names[] = $person['name'];
        }
    }
    return implode(', ', $names);
}

$names = listNames('female', $people);
```

New would be something like:

```php
$names = listNames(filterGender('male'));
```

By doing this, `listNames()` function would *ONLY* care about listing the names and `filterGender()` would *ONLY* care about filtering the _people_ array.

It's important that every function does *ONLY ONE THING*

🌵

**Comment by kristiankostecky** [comment]
Resolved in f780a5c I hope 🙏

**Comment by kristiankostecky** [comment]
Resolved better in cee1678 👍
